misc/package/Data Files/MWSE/core/mcm/fileUtils.lua

```diff
@@ -0,0 +1,71 @@
+-- Fetches Components and Variables by class name.
+-- Intended for internal use only.
+local fileUtils = {}
+
+
+
+local prefixLength = string.len("Data Files\\MWSE\\core\\")
+
+-- Store component/variable paths so we only have to iterate the directory once. 
+-- (Testing has shown this results in a noticeable boost in performance.)
+
+local componentPaths = {}
+local variablePaths = {}
+
+for filePath, dir, fileName in lfs.walkdir("data files\\mwse\\core\\mcm\\components\\") do
+	-- For example, when adding the path of `mwseMCMPage`:
+	-- The key   will be:  "Page" 
+	--        instead of:  "Page.lua"
+	-- The value will be:  "mcm.components.pages.Page" 
+	-- 		  instead of:  "data files\\mwse\\core\\mcm\\components\\pages\\Page.lua"
+	componentPaths[fileName:sub(1, fileName:len() - 4)] = filePath:sub(1 + prefixLength, filePath:len() - 4):gsub("[\\/]", ".")
+end
+
+for filePath, dir, fileName in lfs.walkdir("data files\\mwse\\core\\mcm\\variables\\") do
+	variablePaths[fileName:sub(1, fileName:len() - 4)] = filePath:sub(1 + prefixLength, filePath:len() - 4):gsub("[\\/]", ".")
+end
+
+local components = {}
+local variables = {}
+
+---@protected
+---@param className string The name of the class of the `mwseMCMComponent` to search for.
+---@return mwseMCMComponent?
+function fileUtils.getComponentClass(className)
+	local class = components[className]
+	if class then return class end
+	if className == "HyperLink" then
+		return fileUtils.getComponentClass("Hyperlink")
+	end
+	if className == "SidebarPage" then
+		return fileUtils.getComponentClass("SideBarPage")
+	end
```

Comment:
These things look cursed. Have you considered storing the keys of `componentPaths` and `variablePaths` in lowercase?